eth/stagedsync/stage_mining_exec.go

```diff
@@ -176,6 +211,62 @@ func SpawnMiningExecStage(s *StageState, tx kv.RwTx, cfg MiningExecCfg, quit <-c
 	return nil
 }
 
+func getNextTransactions(
+	cfg MiningExecCfg,
+	tx kv.RwTx,
+	header *types.Header,
+	amount uint16,
+	executionAt uint64,
+) (types.TransactionsStream, error) {
+	txSlots := types2.TxsRlp{}
+	var onTime bool
+	if err := cfg.txPool2DB.View(context.Background(), func(poolTx kv.Tx) error {
+		var err error
+		counter := 0
+		for !onTime && counter < 1000 {
+			remainingGas := header.GasLimit - header.GasUsed
+			if onTime, err = cfg.txPool2.Best(amount, &txSlots, poolTx, executionAt, remainingGas); err != nil {
+				return err
+			}
+			time.Sleep(1 * time.Millisecond)
+			counter++
+		}
+		return nil
+	}); err != nil {
+		return nil, err
+	}
+
+	var txs []types.Transaction //nolint:prealloc
+	for i := range txSlots.Txs {
+		s := rlp.NewStream(bytes.NewReader(txSlots.Txs[i]), uint64(len(txSlots.Txs[i])))
+
+		transaction, err := types.DecodeTransaction(s)
+		if err == io.EOF {
+			continue
+		}
+		if err != nil {
+			return nil, err
+		}
+		if transaction.GetChainID().ToBig().Uint64() != 0 && transaction.GetChainID().ToBig().Cmp(cfg.chainConfig.ChainID) != 0 {
+			continue
+		}
+
+		var sender common.Address
+		copy(sender[:], txSlots.Senders.At(i))
```

Comment:
in-theory can cast next way: https://tip.golang.org/doc/go1.17#language

Reply:
using unsafe.Slice?

Reply:
no "An expression s of type []T may now be converted to array pointer type *[N]T"